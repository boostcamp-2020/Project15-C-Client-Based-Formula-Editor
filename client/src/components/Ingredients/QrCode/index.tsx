import React from 'react';
import QRcode from 'qrcode.react';
import { Dimmer, Loader } from 'semantic-ui-react';

interface QrCodeProps {
  imageUrl: string;
}

function QrCode({ imageUrl }: QrCodeProps) {
  return (
    <>
      <QRcode value={imageUrl} />
      {!imageUrl && (
        <Dimmer active inverted>
          <Loader size="large">Loading</Loader>
        </Dimmer>
      )}
    </>
  );
}

export default QrCode;
