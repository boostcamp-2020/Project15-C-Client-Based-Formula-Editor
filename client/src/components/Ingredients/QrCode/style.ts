import styled from '@emotion/styled';

export const QrCodeContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;
