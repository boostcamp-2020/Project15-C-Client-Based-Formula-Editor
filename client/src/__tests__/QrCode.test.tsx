import * as React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import QrCode from '../components/Ingredients/QrCode/index';
import '@testing-library/jest-dom';

describe('<QrCode />', () => {
  it('renders QrCode', () => {
    const { container } = render(<QrCode imageUrl={'imgUrl'} />);
    expect(container).toMatchSnapshot();
  });
});
