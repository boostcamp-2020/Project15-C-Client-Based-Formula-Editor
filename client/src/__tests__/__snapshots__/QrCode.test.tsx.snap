// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QrCode /> renders QrCode 1`] = `
<div>
  <div
    class="css-1hppjzv"
  >
    <canvas
      height="128"
      style="height: 128px; width: 128px;"
      width="128"
    />
  </div>
</div>
`;
